src/librustdoc/formats/renderer.rs

```diff
@@ -17,6 +17,14 @@ pub(crate) trait FormatRenderer<'tcx>: Sized {
     ///
     /// This is true for html, and false for json. See #80664
     const RUN_ON_MODULE: bool;
+    /// For each module, we go through their items by calling:
+    ///
+    /// 1. make_child_renderer
+    /// 2. set_back_info
+    ///
+    /// This is because since the renderer is not cloned, if a value (module state for example) gets
+    /// updated, we want to reset it to the current module's value.
+    type InfoType;
```

Comment:
The name `InfoType` doesn't say anything about what it it's doing, and neither does this comment.